pkg/apis/security/v1alpha1/types.go

```diff
@@ -35,6 +35,12 @@ type NetworkPolicy struct {
 
 // NetworkPolicySpec defines the desired state for NetworkPolicy.
 type NetworkPolicySpec struct {
+	// Tier specifies the tier to which this NetworkPolicy belongs to.
+	// The NetworkPolicy order will be determined based on the combination of the
+	// Tier's Priority and the NetworkPolicy's own Priority. If not specified,
+	// this policy will be created in the Application Tier right above the K8s
+	// NetworkPolicy which resides at the bottom.
+	Tier string `json:"tier,omitempty"`
```

Comment:
I am curious about why we do not do something like this in this file:
```
type Tier string
const (
    TierEmergency    Tier = "Emergency"
    TierSecurityOps  Tier = "SecurityOps"
    # ...
)
```
Since my understanding based on the openAPI schema for the CRDs is that we constrain the tiers to a predefined set.
This seems to be pretty common in K8s APIs: https://godoc.org/k8s.io/api/core/v1#RestartPolicy

Reply:
i left it as string because idea is to replace the static tiers to CRDs where the name of the tier may not necessarily be part of the enum..